model/src/main/scala/org/broadinstitute/dsde/rawls/model/DataReferenceModel.scala

```diff
@@ -74,6 +74,98 @@ object DataReferenceModelJsonSupport extends JsonSupport {
     }
   }
 
+  implicit object DataReferenceListFormat extends RootJsonFormat[DataReferenceList] {
+    val RESOURCES = "resources"
+
+    override def write(refList: DataReferenceList) = JsObject(
+      RESOURCES -> refList.getResources.asScala.toList.toJson
+    )
+
+    override def read(json: JsValue): DataReferenceList = {
+      json.asJsObject.getFields(RESOURCES) match {
+        case Seq(JsArray(resources)) =>
+          new DataReferenceList().resources(resources.map(_.convertTo[DataReferenceDescription]).asJava)
+        case _ => throw DeserializationException("DataReferenceList expected")
+      }
+    }
+  }
+
+
+  implicit object ResourceMetadataFormat extends RootJsonFormat[ResourceMetadata] {
+    val WORKSPACE_ID = "workspaceId"
+    val RESOURCE_ID = "resourceId"
+    val NAME = "name"
+    val DESCRIPTION = "description"
+    val RESOURCE_TYPE = "resourceType"
+    val STEWARDSHIP_TYPE = "stewardshipType"
+    val CLONING_INSTRUCTIONS = "cloningInstructions"
+
+    override def write(metadata: ResourceMetadata)= JsObject(
+      WORKSPACE_ID -> stringOrNull(metadata.getWorkspaceId),
+      RESOURCE_ID -> stringOrNull(metadata.getResourceId),
+      NAME -> stringOrNull(metadata.getName),
+      DESCRIPTION -> stringOrNull(metadata.getDescription),
+      RESOURCE_TYPE -> stringOrNull(metadata.getResourceType),
+      STEWARDSHIP_TYPE -> stringOrNull(metadata.getStewardshipType),
+      CLONING_INSTRUCTIONS -> stringOrNull(metadata.getCloningInstructions)
+    )
+
+    override def read(json: JsValue): ResourceMetadata = {
+      json.asJsObject.getFields(WORKSPACE_ID, RESOURCE_ID, NAME, DESCRIPTION, RESOURCE_TYPE, STEWARDSHIP_TYPE, CLONING_INSTRUCTIONS) match {
+        case Seq(workspaceId, resourceId, JsString(name), JsString(description), JsString(resourceType), JsString(stewardshipType), JsString(cloningInstructions)) =>
+          new ResourceMetadata()
+            .workspaceId(workspaceId.convertTo[UUID])
+            .resourceId(resourceId.convertTo[UUID])
+            .name(name)
+            .description(description)
+            .resourceType(ResourceType.fromValue(resourceType))
+            .stewardshipType(StewardshipType.fromValue(stewardshipType))
+            .cloningInstructions(CloningInstructionsEnum.fromValue(cloningInstructions))
+      }
+    }
+  }
```

Comment:
the read and write methods here are not symmetrical - the write method can create json containing nulls, but the read method will fail to deserialize those. Any idea if we can expect nulls in practice? At minimum, add code comments and a `case _ => throw DeserializationException(...)`